plugin/pulpcore/plugin/changeset/model.py

```diff
@@ -131,13 +135,42 @@ def save(self):
             except IntegrityError:
                 is_duplicate = True
             else:
-                for artifact in self.artifacts:
-                    artifact.model.content = self.model
-                    artifact.save()
-        if is_duplicate:
-            model = type(self.model)
-            content = model.objects.get(**self.key)
-            self.update(content)
+
+                for remote_artifact in self.artifacts:
+                    if not remote_artifact.content_artifact.artifact_id:
+                        if not deferred:
```

Comment:
The decision to create the artifact should be based on decisions made by the `ArtifactIterator` already.  When the `RemoteArtifact.path` is set, the file has been downloaded and real `Artifact` should be created.